plugins/module_utils/k8s/service.py

```diff
@@ -543,3 +546,77 @@ def hide_field(definition: dict, hidden_field: str) -> dict:
         else:
             del definition[split[0]]
     return definition
+
+
+def decode_response(resp) -> Tuple[Dict, List[str]]:
+    """
+    This function decodes unserialized responses from the Kubernetes python
+    client and decodes the RFC2616 14.46 warnings found in the response
+    headers.
+    """
+    obj = ResourceInstance(None, loads(resp.data.decode("utf8"))).to_dict()
+    warnings = resp.headers["warning"].split(", ") if "warning" in resp.headers else []
```

Comment:
`resp.headers["warning"]` can be `None` so you'll need to make sure there's something there before calling `split()` on it.

Reply:
You mean even though `warning` is in `headers` I should check if `headers["warning"]` returns `None`?

Reply:
Sorry, yeah, `resp.headers` can be `None`. It's a [urllib3 response](https://urllib3.readthedocs.io/en/stable/reference/urllib3.response.html#urllib3.response.HTTPResponse).

Reply:
Updated to address it, does it look good to you now?